catalog/dags/common/sensors/utils.py

```diff
@@ -27,3 +31,30 @@ def get_most_recent_dag_run(dag_id) -> list[datetime] | datetime:
     # ``ExternalTaskSensor::get_count``, especially the handling
     # of ``dttm_filter`` for the relevant implementation details.
     return []
+
+
+@task(retries=0)
+def prevent_concurrency_with_dag(external_dag_id: str, **context):
+    """
+    Prevent concurrency with the given external DAG, by failing
+    immediately if that DAG is running.
+    """
+
+    wait_for_dag = ExternalTaskSensor(
```

Comment:
Can this be implemented using `ExternalDAGSensor` (or a factory function if you replace `ExternalDAGSensor` with one)? The only difference between the two looks like the timeout.